RockLib.Messaging.NamedPipes/NamedPipeSender.cs

```diff
@@ -61,75 +70,53 @@ public Task SendAsync(SenderMessage message, CancellationToken cancellationToken
             };
 
             foreach (var header in message.Headers)
-                namedPipeMessage.Headers.Add(header.Key, header.Value.ToString());
+            {
+                namedPipeMessage.Headers.Add(header.Key, header.Value.ToString()!);
+            }
 
-            var messageString = _serializer.SerializeToString(namedPipeMessage);
-            var completion = new TaskCompletionSource<bool>();
+            var messageString = NamedPipeMessageSerializer.SerializeToString(namedPipeMessage);
 
-            _workItems.Add(new WorkItem { Message = messageString, Completion = completion, CancellationToken = cancellationToken });
+            _workItems.Add(messageString, cancellationToken);
 
-            return completion.Task;
+            return Task.CompletedTask;
         }
 
         /// <summary>
         /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
         /// </summary>
+#pragma warning disable CA1063 // Implement IDisposable Correctly
+#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
         public void Dispose()
+#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
+#pragma warning restore CA1063 // Implement IDisposable Correctly
         {
+            _senderCancellation.Cancel();
+
             _workItems.CompleteAdding();
-            _runThread.Join();
+            _workItems.Dispose();
+
+            _senderCancellation.Dispose();
+
+            if(_sender.IsCompleted)
+            {
+                _sender.Dispose();
+            }
         }
 
-        private void Run()
+        private void Run(CancellationToken token)
         {
-            foreach (var workItem in _workItems.GetConsumingEnumerable())
+            try
             {
-                if (workItem.CancellationToken.IsCancellationRequested)
+                foreach (var workItem in _workItems.GetConsumingEnumerable(token))
                 {
-                    workItem.Completion.SetCanceled();
-                    continue;
-                }
+                    using var pipe = new NamedPipeClientStream(".", PipeName, PipeDirection.Out, PipeOptions.Asynchronous);
+                    pipe.Connect(0);
```

Comment:
I'll take a look again. When I was editing this, I was kind of hacking weeds and I may have cut out too much without realizing it :|.

Reply:
I'm not sure if there was anything being done with the exception with the way it did work. Seems like it would've been ignored anyway.

The `WriteLine()` not having an exception handler....I really don't know what we could do here. If we `try/catch`, we're not going to have any indication of what went wrong.